### PR TITLE
K19 dkpa 58

### PR DESCRIPTION
**Pull request by AlexXDD**
Implemented most of DAO. Project update and merge required for correct work.
Also I'd like to have feedback how to improve my code

### CODE REVIEW

**Comment by AlexXDD** [comment]
I'm about to correct some changes now

**Comment by AlexXDD** [comment]
Changed DB script where role is created. Added "ROLE_APPROVER" with id=2 and moved "ROLE_USER" to id=4 according to AuthirityUtils configuration

**Comment by roman-bessmertnyi** [comment]
codecov says bad. no approve

File: backend/src/main/java/edu/netcracker/backend/controller/ApproverCrudController.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        Optional<User> user = acs.getById(id);
+        return user.orElse(null);
+    }
+
+    @PostMapping("v1/api/admin/approvers")
+    @PreAuthorize("hasAuthority('ROLE_ADMIN')")
+    public void addApprover(@RequestBody User approver) {
+        acs.add(approver);
+    }
+
```
Comment (VityaIvanov) [inline backend/src/main/java/edu/netcracker/backend/controller/ApproverCrudController.java]: Delete empty line

File: backend/src/main/java/edu/netcracker/backend/controller/ApproverCrudController.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    @GetMapping("v1/api/admin/approvers/paging")
+    @PreAuthorize("hasAuthority('ROLE_ADMIN')")
+    public List<User> getApprovers(@RequestParam("limit") Number limit, @RequestParam("offset") Number offset){
+        return acs.getApprovers(limit, offset);
+    }
+
+    @GetMapping("v1/api/admin/approvers")
+    @PreAuthorize("hasAuthority('ROLE_ADMIN')")
+    public User getApprover(@RequestParam("id") Number id) {
+        Optional<User> user = acs.getById(id);
```
Comment (VityaIvanov) [inline backend/src/main/java/edu/netcracker/backend/controller/ApproverCrudController.java]: It is better do work with Optional in the service layer. In controller you must work with User

File: backend/src/main/java/edu/netcracker/backend/controller/ApproverCrudController.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    @GetMapping("v1/api/admin/approvers/paging")
+    @PreAuthorize("hasAuthority('ROLE_ADMIN')")
+    public List<User> getApprovers(@RequestParam("limit") Number limit, @RequestParam("offset") Number offset){
+        return acs.getApprovers(limit, offset);
+    }
+
+    @GetMapping("v1/api/admin/approvers")
+    @PreAuthorize("hasAuthority('ROLE_ADMIN')")
+    public User getApprover(@RequestParam("id") Number id) {
```
Comment (VityaIvanov) [inline backend/src/main/java/edu/netcracker/backend/controller/ApproverCrudController.java]: You can use UserDTO instead User

**Pull request merged**